src/core/algorithms/nd/nd.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include "model/table/column.h"
+#include "model/table/vertical.h"
+
+namespace model {
+
+using WeightType = unsigned;
```

Comment:
It is better to specify the full name of the type. In this case it is better to use `unsigned int`.